federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -114,6 +125,15 @@ func Run(s *options.CMServer) error {
 		glog.Fatal(server.ListenAndServe())
 	}()
 
+	federationClientset, err := federationclientset.NewForConfig(restclient.AddUserAgent(restClientCfg, "federation-controller-manager"))
+	if err != nil {
+		glog.Fatalf("Invalid API configuration: %v", err)
+	}
+
+	if err := createFederationNamespace(federationClientset, federationOnlyNamespace); err != nil {
```

Comment:
Agree with the concern, but even in case of non-HA federation control plane, it has one purpose to wait for the Api Server and to establish it is working fine. In the current scenario, without this. the federation-controller-manager dies and another pod is brought up. This is documented in this issue https://github.com/kubernetes/kubernetes/issues/44490